source/processes/fs.cpp

```diff
@@ -2747,10 +2782,72 @@ std::tuple<OS::Result, IPC::MappedBuffer>
 FakeFS::HandleCreateExtSaveData(FakeThread& thread, ProcessId, const Platform::FS::ExtSaveDataInfo& info,
                                 uint32_t max_directories, uint32_t max_files, uint64_t, uint32_t,
                                 IPC::MappedBuffer input_smdh) {
-    auto result = ArchiveExtSaveData::CreateSaveData(thread, *this, info, max_directories, max_files, true);
+    auto result = ArchiveExtSaveData::CreateSaveData(thread, *this, info, max_directories, max_files, input_smdh, true);
     return std::make_tuple(result, input_smdh);
 }
 
+std::tuple<OS::Result, uint32_t, IPC::MappedBuffer>
+FakeFS::HandleEnumerateExtSaveData( FakeThread& thread, ProcessId, uint32_t, uint32_t media_type,
+                                    uint32_t id_entry_size, uint32_t is_shared, IPC::MappedBuffer output_ids) {
+    // Game carts cannot have extdata
+    if (media_type == 2)
+        return std::make_tuple(0xE0C046FA, 0, output_ids);
+
+    if (id_entry_size != 4 && id_entry_size != 8 || output_ids.size % id_entry_size)
+        return std::make_tuple(0xE0E046BC, 0, output_ids);
+
+    std::filesystem::path extdata_high_dir;
+    uint32_t extdata_id_high = 0;
+
+    std::tie(extdata_high_dir, extdata_id_high) = std::invoke([&]() -> std::tuple<std::filesystem::path, uint32_t> {
+        std::filesystem::path extdata_root = ArchiveExtSaveData::GetExtDataRootDirectory(*this, static_cast<MediaType>(media_type));
+        if (media_type == 0) {
+            if (!is_shared)
+                throw std::runtime_error("NAND extdata must be shared");
+            return std::make_tuple(extdata_root / "00048000", 0x00048000);
+        }
+
+        if (is_shared)
+            throw std::runtime_error("SD extdata must not be shared");
+
+        return std::make_tuple(extdata_root / "00000000", 0x00000000);
+    });
+
+    uint32_t num_written = 0;
+    uint32_t buf_remain = output_ids.size / id_entry_size;
+
+    std::function<void(uint64_t)> add_extdata_id;
+    if (id_entry_size == 4)
+        add_extdata_id = [&](uint64_t ext_id) -> void {
+            thread.WriteMemory32(output_ids.addr + (num_written * 4), ext_id);
+        };
+    else
+        add_extdata_id = [&](uint64_t ext_id) -> void {
+            thread.WriteMemory32(output_ids.addr + (num_written * 8), ext_id);
+            thread.WriteMemory32(output_ids.addr + (num_written * 8) + 4, ext_id >> 32);
+        };
+
+    for (auto const& extdata_low_dir : std::filesystem::directory_iterator{extdata_high_dir})
+    {
+        if (!buf_remain)
```

Comment:
If we rename this variable to `max_entries`, we can just check `max_entries == num_written` here and hence only need one counter variable instead of two.